### PR TITLE
Helmholtz login improvements

### PR DESCRIPTION
**Pull request by ruester**
# Helmholtz login improvements

Fixes https://github.com/research-software-directory/Helmholtz/issues/16

Changes proposed in this pull request:

* as we realized that also social providers and many international institutes are listed for Helmholtz AAI login we want to make sure that only Helmholtz Centres can login
* by default only Helmholtz centres listed in https://hifis.net/doc/helmholtz-aai/list-of-vos/ can login
* with the environment variable `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` set to `true` all logins via Helmholtz AAI are allowed, including non-Helmholtz institutes and social IdPs

How to test:

* set `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` to `true`, go to frontend website `localhost/login`, click on Helmholtz AAI logo and log in with Google or ORCID
* set `HELMHOLTZAAI_ALLOW_EXTERNAL_USERS` to `false` or comment out, login again with Google or ORCID and it should not work

Unfortunately I did not find a way to send the frontend the reason why the login did not work. Because it may did not go wrong, but the user is just not allowed to login since the RSD is configured this way.

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [x] Link to a GitHub issue
*   [ ] Update documentation
*   [x] Tests

### CODE REVIEW

File: authentication/src/main/java/nl/esciencecenter/rsd/authentication/Config.java [hunk 1]
```diff
@@ -63,4 +63,15 @@ public static String helmholtzAaiTokenUrl() {
 	public static String helmholtzAaiScopes() {
 		return System.getenv("NEXT_PUBLIC_HELMHOLTZAAI_SCOPES");
 	}
+
+	public static boolean helmholtzAaiAllowExternalUsers() {
```
Comment (ewan-escience) [inline authentication/src/main/java/nl/esciencecenter/rsd/authentication/Config.java:67]: You can replace all the lines with `return Boolean.parseBoolean(System.getenv("HELMHOLTZAAI_ALLOW_EXTERNAL_USERS"));`. The method [parseBoolean](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Boolean.html#parseBoolean(java.lang.String)) will take care of emty strings and nulls

File: authentication/src/test/java/nl/esciencecenter/rsd/authentication/HelmholtzAaiLoginTest.java [hunk 2]
```diff
@@ -18,7 +18,43 @@ void testOrganisationFromEntitlementsExpected() {
...
+		arr.add("urn:geant:helmholtz.de:group:Helmholtz-member#login-dev.helmholtz.de");
+		arr.add("urn:geant:helmholtz.de:group:GFZ#login-dev.helmholtz.de");
+		arr.add("urn:mace:dir:entitlement:common-lib-terms");
+
+		assertEquals(
+			"GFZ",
+			HelmholtzAaiLogin.getOrganisationFromEntitlements(arr, true)
+		);
+
+		assertEquals(
```
Comment (ewan-escience) [inline authentication/src/test/java/nl/esciencecenter/rsd/authentication/HelmholtzAaiLoginTest.java:48]: You could use [assertNull](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertNull-java.lang.Object-) instead.

**Review by ewan-escience — APPROVED** [review]
As far as I could tell it all works. I left two small suggestions with code improvements. And thanks for noticing the `cosumed` typo. :)

File: authentication/src/test/java/nl/esciencecenter/rsd/authentication/HelmholtzAaiLoginTest.java (hunk 2 above)
Comment (ruester) [reply authentication/src/test/java/nl/esciencecenter/rsd/authentication/HelmholtzAaiLoginTest.java:48]: yes, much better, thanks!

File: authentication/src/main/java/nl/esciencecenter/rsd/authentication/Config.java (hunk 1 above)
Comment (ruester) [reply authentication/src/main/java/nl/esciencecenter/rsd/authentication/Config.java:67]: thanks!

**Comment by ruester** [comment]
Implemented the requested changes now and pushed the branch again :+1:

**Pull request merged**